src/Components/Form/Form.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react';
+import ApiRequest from '../ApiRequest/ApiRequest';
+import './Form.scss';
+
+class Form extends Component {
+  submitForm = event => {
+    event.preventDefault();
+
+    const locationFieldValue = event.target.location.value;
+    const data = ApiRequest(locationFieldValue);
+
+    fetch(data)
+      .then(response => response.json())
+      .then(response => this.props.handleState(response))
```

Comment:
in this line you already got an json not a response itself so change name to json 
`.then(json -> onFetch(json))`

as I mentioned handleState is wrong. 

`onFetch `will be better name for this.
and also remember to DESTRUCTURE from props. I said it already 100 times and still the same issue, Please focus :)

Reply:
I named that as setWheaterDatacuz on App.js I'm pasting method in to deeper component to use it.